File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/handler/WebSrcImageCopyUrlHandler.java
package jp.hazuki.yuzubrowser.webkit.handler;

import android.content.Context;

import java.lang.ref.WeakReference;

import jp.hazuki.yuzubrowser.utils.ClipboardUtils;

public class WebSrcImageCopyUrlHandler extends WebSrcImageHandler {
    private final WeakReference<Context> mReference;

    public WebSrcImageCopyUrlHandler(Context context) {
        mReference = new WeakReference<>(context);
    }

    @Override
    public void handleUrl(String url) {
        Context context = mReference.get();
        if (context != null)
            ClipboardUtils.setClipboardText(context, url);
    }
}

File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/handler/WebSrcImageCopyUrlHandler.kt
package jp.hazuki.yuzubrowser.webkit.handler

import android.content.Context
import jp.hazuki.yuzubrowser.utils.extensions.setClipboardWithToast
import java.lang.ref.WeakReference

class WebSrcImageCopyUrlHandler(context: Context) : WebSrcImageHandler() {
    private val mReference: WeakReference<Context> = WeakReference(context)

    override fun handleUrl(url: String) {
        mReference.get()?.run {
            setClipboardWithToast(url)
        }
    }
}
